substrate/runtime/staking/src/contract.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Smart-contract execution module.
```

Comment:
a little weird to have this in the general "staking" module as opposed to its own crate. we don't actually want generalized smart contracts in the Polkadot runtime, although we do want staking. not a blocker for this PR but I would prefer a split in the future

Reply:
that's totally makes sense! I will try to figure it out in the following PRs